### ISSUE TITLE
Where to set training parameter in OBB

### ISSUE THREAD

**Issue by Tian14267** [ultralytics/ultralytics#21468]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I want to know ,where can i set training parameter in **OBB** task, like  `lr `or other parameter .

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Tian14267, thank you for reaching out and for your interest in Ultralytics 🚀! This is an automated response to help you get started—an Ultralytics engineer will review your question and provide further assistance soon.

We recommend checking out the [Docs](https://docs.ultralytics.com/) for helpful guidance. Specifically, you can find [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples, which may cover common questions about configuring training parameters for tasks like OBB.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question (like setting parameters for OBB), please include as much information as you can, such as your training command, configuration files, and any logs. Make sure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our community for more real-time support:
- [Discord](https://discord.com/invite/ultralytics) 🎧
- [Discourse](https://community.ultralytics.com/)
- [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Upgrade

To ensure the latest features and fixes are available, please upgrade the `ultralytics` package and its [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your question—our team will follow up with more details soon!

**Comment by Y-T-G** [comment]
You pass all the parameters to `model.train()`

https://docs.ultralytics.com/modes/train/#__tabbed_1_1

`results = model.train(..., lr=0.001)`

**Comment by Tian14267** [comment]
> model.train

Only can set parameters  in `model.train` ？ This feels too troublesome.

An other question, I  get two box predict in one target. Like this:

<img width="1032" height="776" alt="Image" src="https://github.com/user-attachments/assets/a8e8a574-0c0b-4de6-a7db-7441d1957064" />

Does anybody know how to solve this ?

**Comment by Y-T-G** [comment]
You can use `agnostic_nms=True` during prediction

**Comment by Tian14267** [comment]
> You can use `agnostic_nms=True` during prediction

Thank you very much.

**Comment by glenn-jocher** [comment]
You're welcome! Regarding training parameters, passing them to `model.train()` is actually the standard approach across all YOLO tasks, not just OBB. This design keeps the interface consistent and flexible.

If you prefer setting default parameters, you can also create a custom YAML configuration file and pass it via the `cfg` parameter, though `model.train()` arguments will override any YAML settings when specified.
